contrib/platform/src/com/sun/jna/platform/win32/Wbemcli.java

```diff
@@ -0,0 +1,172 @@
+ * Apache License 2.0. (starting with JNA version 4.0.0).
+ * 
+ * You can freely decide which license you want to apply to 
+ * the project.
+ * 
+ * You may obtain a copy of the LGPL License at:
+ * 
+ * http://www.gnu.org/licenses/licenses.html
+ * 
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "LGPL2.1".
+ * 
+ * You may obtain a copy of the Apache License at:
+ * 
+ * http://www.apache.org/licenses/
+ * 
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "AL2.0".
+ */
+package com.sun.jna.platform.win32;
+
+import com.sun.jna.NativeLong;
+import com.sun.jna.Pointer;
+import com.sun.jna.platform.win32.Guid.CLSID;
+import com.sun.jna.platform.win32.Guid.GUID;
+import com.sun.jna.platform.win32.Variant.VARIANT;
+import com.sun.jna.platform.win32.WTypes.BSTR;
+import com.sun.jna.platform.win32.WinNT.HRESULT;
+import com.sun.jna.platform.win32.COM.COMUtils;
+import com.sun.jna.platform.win32.COM.Unknown;
+import com.sun.jna.ptr.LongByReference;
+import com.sun.jna.ptr.PointerByReference;
+
+/**
+ * This header is used by Remote Desktop Services.
+ */
+public interface Wbemcli {
+
+    public static final int WBEM_FLAG_RETURN_IMMEDIATELY = 0x00000010;
+    public static final int WBEM_FLAG_FORWARD_ONLY = 0x00000020;
+    public static final int WBEM_INFINITE = 0xFFFFFFFF;
+
+    // Non-error constants
+    // https://docs.microsoft.com/en-us/windows/desktop/wmisdk/wmi-non-error-constants
+    public static final int WBEM_S_NO_ERROR = 0x0;
+    public static final int WBEM_S_FALSE = 0x1;
+    public static final int WBEM_S_TIMEDOUT = 0x40004;
+    public static final int WBEM_S_NO_MORE_DATA = 0x40005;
+
+    // Error constants
+    // https://docs.microsoft.com/en-us/windows/desktop/wmisdk/wmi-error-constants
+    public static final int WBEM_E_INVALID_NAMESPACE = 0x8004100e;
+    public static final int WBEM_E_INVALID_CLASS = 0x80041010;
+    public static final int WBEM_E_INVALID_QUERY = 0x80041017;
+
+    /**
+     * Holds a row of results of a WMI query
+     */
+    class IWbemClassObject extends Unknown {
+
+        public IWbemClassObject(Pointer pvInstance) {
+            super(pvInstance);
+        }
+
+        public HRESULT Get(BSTR wszName, NativeLong lFlags, VARIANT.ByReference pVal, Pointer pvtType,
+                LongByReference plFlavor) {
+            // Get is 5th method of IWbemClassObjectVtbl in WbemCli.h
+            return (HRESULT) _invokeNativeObject(4,
+                    new Object[] { getPointer(), wszName, lFlags, pVal, pvtType, plFlavor }, HRESULT.class);
+        }
+    }
+
+    /**
+     * Iterates to the next row of results of a WMI query
+     */
+    class IEnumWbemClassObject extends Unknown {
+
+        public IEnumWbemClassObject(Pointer pvInstance) {
+            super(pvInstance);
+        }
+
+        public HRESULT Next(NativeLong lTimeOut, NativeLong uCount, PointerByReference ppObjects,
```

Comment:
Same comment as in `IWbemClassObject` applies - is it expected, that this method is called on a non windows platform? if not, `NativeLong` can be substituted with `int`.